ha/monitor/k8s/monitor.py

```diff
@@ -51,36 +51,62 @@ def __init__(self, wait_time=10):
             _conf_stor_search = ConftStoreSearch()
 
             self.monitors = []
+            monitor_args = {}
 
             # event output in pretty format
-            kwargs = {K8SClientConst.PRETTY : True}
+            monitor_args['watch_args'] = {K8SClientConst.PRETTY : True}
 
-            # Seting a timeout value, 'timout_seconds', for the stream.
+            # Setting a timeout value, 'timeout_seconds', for the stream.
             # timeout value for connection to the server
             # If do not set then we will not able to stop immediately,
-            # becuase synchronus function watch.stream() will not come back
+            # because synchronous function watch.stream() will not come back
             # until catch any event on which it is waiting.
-            kwargs[K8SClientConst.TIMEOUT_SECONDS] = K8SClientConst.VAL_WATCH_TIMEOUT_DEFAULT
+            monitor_args['watch_args'][K8SClientConst.TIMEOUT_SECONDS] = wait_time
 
             # Get MessageBus producer object for all monitor threads
             producer = self._get_producer()
 
             # Change to multiprocessing
             # Creating NODE monitor object
-            node_monitor = ObjectMonitor(producer, K8SClientConst.NODE, **kwargs)
+            Log.info("Instantiating monitor for all the nodes in cluster.")
+            node_monitor = ObjectMonitor(producer, K8SClientConst.NODE, **monitor_args)
             self.monitors.append(node_monitor)
 
-            _, nodes_list = _conf_stor_search.get_cluster_cardinality()
-            if not nodes_list:
-                Log.warn(f"No nodes in the cluster to watch for nodes_list: {nodes_list}")
+            _, label_id_map, resource_id_map = _conf_stor_search.get_cluster_cardinality()
+            Log.debug(f"label id map: {label_id_map}, resource id map: {resource_id_map}")
+
+            # NOTE: Currently running two pod monitors with the label 'cortx.io/machine-id' and
+            #  'statefulset.kubernetes.io/pod-name'. in any deployment, both labels will not co-exist,
+            #  so either one pod monitor will not be receiving any events.
+            # TODO: CORTX-31875 So once the label 'statefulset.kubernetes.io/pod-name' is available,
+            #  the pod monitor with the label 'cortx.io/machine-id' needs to be removed as,
+            #  it is running only for backward compatibility.
+            # 1. pod monitor for pods with 'cortx.io/machine-id' labels
+            if label_id_map:
```

Comment:
This init function looks huge, is it possible to break it down into multiple functions, would be easier to read.

Reply:
I tried but can't as this function does nothing but create 3 monitor objects.
and 1 pod monitor object creation code and big comment are to be removed (TODO) so anyway this function will be small enough.
let me know your thought on this.